### PR TITLE
Add RestrictedToInstruments to OrderItem.

### LINKED ISSUE #408: RestrictedToInstruments property is missing
Hi. I can not find the [RestrictedToInstruments property](https://developer.swedbankpay.com/payment-menu/payment-order#step-1-create-payment-order) of OrderItem (Create Payment Order Request). How can I specify instruments in my request?

### PR DESCRIPTION
**Pull request by joakimmag**
Fixes #408

### CODE REVIEW

**Review by asbjornu — CHANGES_REQUESTED** [review]
Thanks for implementing this, but the `paymentOrder.restrictedToInstruments` field is deprecated. (Not to be confused with `orderItem.restrictedToInstruments`.) It should now be `instrument` instead. You can [read more about how `instrument` works in the documentation](https://developer.swedbankpay.com/payment-menu/features/optional/instrument-mode).

**Comment by wireless-r** [comment]
> Thanks for implementing this, but the `paymentOrder.restrictedToInstruments` field is deprecated. (Not to be confused with `orderItem.restrictedToInstruments`.) It should now be `instrument` instead. You can [read more about how `instrument` works in the documentation](https://developer.swedbankpay.com/payment-menu/features/optional/instrument-mode).

I implemented [`orderItem.restrictedToInstruments`](https://developer.swedbankpay.com/payment-menu/payment-order#payment-url) here.

**Comment by wireless-r** [comment]
> Thanks for implementing this, but the `paymentOrder.restrictedToInstruments` field is deprecated. (Not to be confused with `orderItem.restrictedToInstruments`.) It should now be `instrument` instead. You can [read more about how `instrument` works in the documentation](https://developer.swedbankpay.com/payment-menu/features/optional/instrument-mode).

I implemented [`orderItem.restrictedToInstruments`](https://developer.swedbankpay.com/payment-menu/payment-order#payment-url) here.

File: src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs [hunk 1]
```diff
@@ -25,6 +26,7 @@ internal OrderItem(OrderItemDto dto)
             Description = dto.Description;
             DiscountDescription = dto.DiscountDescription;
             DiscountPrice = dto.DiscountPrice;
+            RestrictedToInstruments = dto.RestrictedToInstruments as IList<OrderItemInstrument>;
```
Comment (asbjornu) [inline src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs:29]: Although this conversion through an implicit operator is clever, I like when things are more explicit. I would also like to have at least one test that ensures this conversion works in place.

**Review by asbjornu — CHANGES_REQUESTED** [review]
> I implemented [`orderItem.restrictedToInstruments`](https://developer.swedbankpay.com/payment-menu/payment-order#payment-url) in this pull request. The field is shown in the example request – I think its purpose is only to add the order item if you pay by the specified payment method.

That is indeed correct, sorry for misreading the PR. 🤦🏼

File: src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs (hunk 1 above)
Comment (wireless-r) [reply src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs:29]: Thanks for your feedback. Before I go ahead and write test - would this be better, more explicit?
```cs
RestrictedToInstruments = dto.RestrictedToInstruments?.Select(instrument => OrderItemInstrument.FromValue(instrument)).ToList();
```

Comment (asbjornu) [reply src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs:29]: Yes, that looks good to me. 👍🏼

Comment (wireless-r) [reply src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs:29]: For test, I could write a method that:

1. Creates payment order with one order item with `RestrictedToInstruments` values. Will add a few values.
2. Checks the response if the same `RestrictedToInstruments` values can be found.

Would it be good?

Comment (wireless-r) [reply src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs:29]: For test, I can write a method that:

1. Creates payment order with one order item with `RestrictedToInstruments` property. 
2. Checks the response order item if it has `RestrictedToInstruments` property with the same items.

If it passes, it means both request and response map correctly.

Comment (asbjornu) [reply src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs:29]: Yep, that sounds good! 👍🏼

Comment (wireless-r) [reply src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs]: I have updated this pull request with the changes.

**Review by asbjornu — APPROVED** [review]

**Comment by joakimmag** [comment]
I apparently did a merge attempt - it was but a misclick.

The test job failed:
> SwedbankPay.Sdk.Exceptions.HttpResponseException : POST: https://api.externalintegration.payex.com/psp/trustly/payments?$expand=prices,payeeinfo,urls,transactions,authorizations,captures,reversals,cancellations,sales,metadata failed with error code **Unauthorized using bearer token rememberToSetThisWhenTesting**

I guess I should not upload my API token and merchant ID? How can I make it pass?

**Comment by asbjornu** [comment]
@joakimmag, pull requests from forks don't have access to secrets for security reasons. We are going to tweak the workflow so the tests aren't run on pull requests from forks.

**Pull request merged**